### DISCUSSION TITLE
HW1 parts

### DISCUSSION THREAD

**Discussion by ours1011** [sanni/cartreader#859] [Need help]
Hi there,

I found parts of a kit I've never built back in the day.

![20231011_172752](https://github.com/sanni/cartreader/assets/76648075/f3dd6dda-2a46-4eb7-a03e-26288df7b612)

According to the wiki it looks like it is V1 of the cart reader, unfortunately there are no infos available below V2 on the wiki.
I also tried the wayback machine with no success :(

I've already ordered a Si5351A clock generator and an Arduino MEGA2560 MEGA16U2.

Is the screen the same as the V3 reader ?
I have a PIC12F629 and means to flash it.
10k resistor is ok.
Which values should I use for the RGB resistors of the LED ? (220 ohm ?)

What else do I need ?

Thanks !

**Comment by Ancyker** [comment]
HW1/V1 is a true shield for the full-size mega. You can probably find info on it on the Arduino forum as that's where the project started.

I believe the LCD is the same, looks like it at least.

You can use any [LED resistor calculator](https://www.digikey.com/en/resources/conversion-calculators/conversion-calculator-led-series-resistor) to calculate the needed values for your particular LEDs as it usually varies by color. You can choose to use the highest value that the calculator returns if you want to minimize part number count. Higher values will cause the LED to be dimmer than rated, however. Alternatively, you can YOLO it and just use 300-1k ohms for everything (again, higher value = dimmer LED).

You can [use the picture on the wiki](https://github.com/sanni/cartreader/wiki/Cart-Reader-V1) for a general idea of how to assemble it. There aren't many components, so it does show most of what you need to do.

**Comment by ours1011** [comment]
Thanks for your comment :)

I've seen the picture over [here](https://github.com/sanni/cartreader/wiki/Cart-Reader-V1) unfortunately it doesn't help much about the values or parts numbers.
I've also found pictures over [there](https://github.com/sanni/cartreader/wiki/About), especially [this one.](https://camo.githubusercontent.com/15d13d9f839be3ad6daf1d643b1654312d929d7cd1065742a231e7421bb743b2/68747470733a2f2f646c2e64726f70626f7875736572636f6e74656e742e636f6d2f732f697334796f38746d317472387866732f76657273696f6e73253230253238342532392e4a50473f646c3d31)

Also do I need jumpers on the "CLK0 CLK1 CLK CPUCLK" ?

**Reply by Ancyker** [reply]
You just need a header there to use with the clockgen, similar to the other revisions. The pins appear to be a 1:1 map with it:

![image](https://github.com/sanni/cartreader/assets/7105333/224add41-3cbd-4f66-a70a-24ffe94225f3)

**Comment by sanni** [comment]
I found some HW1 related stuff on my PC and added it to the [Wiki](https://github.com/sanni/cartreader/wiki/Cart-Reader-V1#more-information)

As an alternative you can also order a [HW3 PCB](https://github.com/sanni/cartreader/wiki/Cart-Reader-V3) and build that since most of the parts between HW1 and HW3 are the same and HW3 has better documentation.

**Reply by ours1011** [reply]
Thanks a lot !
I see you've removed the text that said a line had to be changed in the code to run on HW1, am I correct ?

My goal is to be able to dump and reflash SF Memory (SFC) carts, do I need to have the PIC12F629 onboard or not ?

**Reply by sanni** [reply]
You don't need the PIC/snesCIC for SF Memory carts, only the clock generator.

In config.h you have to change "//#define HW1" to "#define HW1" ([line 36](https://github.com/sanni/cartreader/blob/master/Cart_Reader/Config.h#L36)) 
and "//#define clockgen_installed" to "#define clockgen_installed" ([line 56](https://github.com/sanni/cartreader/blob/master/Cart_Reader/Config.h#L56))

Then flash the Arduino like explained here: https://github.com/sanni/cartreader/wiki/How-to-flash-the-Arduino

**Reply by ours1011** [reply]
Thanks !
I'll do that once I get the parts (Arduino / clock / screen)

**Comment by ours1011** [comment]
I finally got the CLK Generator and the OLED screen, I managed to solder everything together
For now it is with wires mainly because I didn't have female pin headers but also because I want to give it a casing :)
![20231021_231917](https://github.com/sanni/cartreader/assets/76648075/2d82232a-3cb0-4770-b77e-8074800b4296)

The SF Memory I've bought was labeled as Empty and it was
![20231021_234048](https://github.com/sanni/cartreader/assets/76648075/5736a48b-91d1-4f50-8f25-045debd290f6)

The translated text scrolling in the bottom reads as such :
![Capture](https://github.com/sanni/cartreader/assets/76648075/feb4f6be-756f-425b-8a10-755facc245a4)

Everything went smoothly with flashing !

And now the cart isn't empty anymore :)
![20231022_004253](https://github.com/sanni/cartreader/assets/76648075/809e6a42-d396-4e09-8c71-e6085d9fb7f5)

This beast really delivers ! Thanks @sanni and everybody involved in this !

Next steps will be the PIC and building the N64 adapter :)

**Comment by Chomemel** [comment]
Very well done Ours.
Can't wait to see it with the casing.
N'hesite pas à poster des retours sur U-C aussi, y a un genre de topic dédié.
